prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -937,7 +964,10 @@ ssize_t rxr_pkt_proc_matched_rtm(struct rxr_ep *ep,
 
 	pkt_type = rxr_get_base_hdr(pkt_entry->pkt)->type;
 
-	if (pkt_type == RXR_LONG_MSGRTM_PKT || pkt_type == RXR_LONG_TAGRTM_PKT)
+	if (pkt_type == RXR_LONG_MSGRTM_PKT ||
+	    pkt_type == RXR_LONG_TAGRTM_PKT ||
+	    pkt_type == RXR_DC_LONG_MSGRTM_PKT ||
+	    pkt_type == RXR_DC_LONG_TAGRTM_PKT)
```

Comment:
This is a little confusing, why for DC_LONG we get tx_id from ```rxr_get_long_rtm_base_hdr``` but other case we get from ```rxr_get_dc_rtm_base_hdr```?

Reply:
For other dc hdr types, the `tx_id` field is right after `struct rxr_dc_rtm_base_hdr hdr`. 

For `struct rxr_long_rtm_base_hdr`, there is a field `uint64_t data_len` between `tx_id` and `struct rxr_rtm_base_hdr hdr`

Since `struct rxr_long_rtm_base_hdr` already has the field `tx_id`. I think the cost of defining a new type of dc hdr type whose `tx_id` field is right after `struct rxr_dc_rtm_base_hdr hdr` is higher than that of using the existing `struct rxr_long_rtm_base_hdr` and do a branch prediction here.

Reply:
In this case the name ```rxr_get_dc_rtm_base_hdr``` is misleading, the name would indicate all dc rtm has it. Maybe it should be named something else?

Reply:
I wouldn't worry about adding another branch for this here, we already have so many. Let's go for clarity and I agree with Wei that it's not clear.

Reply:
Reimplemented the branches: https://github.com/ofiwg/libfabric/pull/6205/files#diff-406e99ade0d03e770540fc56cc444ed448d046cd3065b645f54f791e05c7b6f9R988